packages/core/src/service/services/product-option-group.service.ts

```
@@ -223,4 +335,135 @@ export class ProductOptionGroupService {
+        if (!hasPermission) {
+            throw new ForbiddenError();
+        }
+        const groupsToAssign = await this.connection.getRepository(ctx, ProductOptionGroup).find({
+            where: {
+                id: In(input.groupIds),
+                productId: IsNull(),
+            },
+            relations: ['options'],
+        });
+        const optionsToAssign = groupsToAssign.reduce(
+            (values, group) => [...values, ...group.options],
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)